packages/vm/src/compatability.rs

```diff
@@ -40,11 +42,46 @@ pub fn check_wasm(wasm_code: &[u8]) -> Result<()> {
             ));
         }
     };
+    check_wasm_memories(&module)?;
     check_wasm_exports(&module)?;
     check_wasm_imports(&module)?;
     Ok(())
 }
 
+fn check_wasm_memories(module: &Module) -> Result<()> {
+    let section = match module.memory_section() {
+        Some(section) => section,
+        None => {
+            return make_validation_err("Wasm contract doesn't have a memory section".to_string());
```

Comment:
Is this an error? Or the same as a section with undefined memory?

What do our sample contracts look like (in memory section)?

Reply:
There is no fixed memory sestion. A Wasm file is a list of sections, each having a type. The wat `(module)` for example does not have a memory section.

Out contract has the sections:

```
$ wasm-objdump --headers packages/vm/testdata/contract.wasm


contract.wasm:	file format wasm 0x1

Sections:

     Type start=0x0000000a end=0x00000076 (size=0x0000006c) count: 16
   Import start=0x00000078 end=0x000000da (size=0x00000062) count: 5
 Function start=0x000000dd end=0x000001c3 (size=0x000000e6) count: 228
    Table start=0x000001c5 end=0x000001ca (size=0x00000005) count: 1
   Memory start=0x000001cc end=0x000001cf (size=0x00000003) count: 1
   Global start=0x000001d1 end=0x000001ea (size=0x00000019) count: 3
   Export start=0x000001ec end=0x00000259 (size=0x0000006d) count: 9
     Elem start=0x0000025c end=0x000002f1 (size=0x00000095) count: 1
     Code start=0x000002f5 end=0x0001cfe7 (size=0x0001ccf2) count: 228
     Data start=0x0001cfea end=0x00020555 (size=0x0000356b) count: 42
```

Reply:
https://github.com/CosmWasm/cosmwasm/pull/262/commits/7e10017e4018d49473f58c9202cac757c1484e42 adds a test for a Wasm with memory section that has 0 memory elements.